### DISCUSSION TITLE
Implementing Autocompletion for Registered Names in a React Project using TypeScript

### DISCUSSION THREAD

**Discussion by kkroeger93** [react-hook-form/react-hook-form#9886] [Q&A]
Hey, 
_my question might be a little out of Scope, if so I'm sorry and you can close the question._
 
How can I create a simple `register` function (similar to the register() method hook form provides) in a React project that accepts a `name` property and provides autocompletion for the registered names in a code editor or IDE, such as within a `unregister` method?

Cheers and thanks a lot :)

**Comment by bluebill1049** [comment]
I think it's better to Controller, as your Autocompletion component is most likely to be controlled.
